autoload/picker.vim

```diff
@@ -13,6 +13,40 @@ function! s:InGitRepository() abort
     return v:shell_error == 0
 endfunction
 
+function! s:EchoLines(lines) abort
+    " Given a list of strings, construct an echo-like shell command that will
+    " print all strings without any shell expansion shenanigans, but with each
+    " string separated by a linefeed character (aka \n).
+    "
+    " There is a concern that the length of the shell command may exceed the
+    " limits imposed by the shell.
+    "
+    " Parameters
+    " ----------
+    " lines : List String
+    "     Strings to be composed into an echo-like shell command with newlines
+    "     separating each string.
+    "
+    " Returns
+    " -------
+    " String
+    "     An echo-like shell command
+    " vint: -ProhibitUnnecessaryDoubleQuote
+    let l:command = "printf '"
+    " Add a %s\\n for each line.
+    let l:command .= repeat('%s\n', len(a:lines))
+    let l:command .= "' "
+    " vint: +ProhibitUnnecessaryDoubleQuote
+    " Add shell escaped lines as arguments.
+    if &shell =~# 'fish'
+      "fish will still try to evaluate backslashes, so escape all the things.
+      let l:command .= join(map(a:lines, 'shellescape(escape(v:val, "\\"))'))
```

Comment:
I wanted to avoid shell specific fixes, but I can't find a consistent tool across bash, zsh, and fish to achieve this.

Apparently, I goofed on my previous testing (though I can't determine what I screwed up), because fish should have generated different results in that screenshot.  Adding the extra escape for backslashes produces results consistent with my three-shell screenshot from earlier.